src/hooks/useBlockList.ts

```diff
@@ -0,0 +1,107 @@
+import ConnectyCube from "connectycube";
+import { useEffect, useState } from "react";
+
+export const BLOCK_LIST_NAME = "@ConnectyCube/blockList";
+
+export type BlockListHook = {
+  blockedUsers: Set<number>;
+  isBlockedUser: (userId: number) => boolean;
+  unblockUser: (userId: number) => Promise<void>;
+  blockUser: (userId: number) => Promise<void>;
+};
+
+enum BlockAction {
+  ALLOW = "allow",
+  DENY = "deny",
+}
+
+function useBlockList(isConnected: boolean): BlockListHook {
+  const [state, setState] = useState<Set<number>>(new Set<number>());
+
+  const addToState = (userId: number): void => {
+    setState((state) => {
+      state.add(userId);
```

Comment:
React state should be treated as immutable. In your code, you're directly mutating the existing Set by calling its .add() method, and then returning the same reference. This can lead to issues because React relies on detecting changes via new object references. If the state reference remains unchanged, React might not trigger a re-render.
 
 Let's replaced to
 
 ```
 const addToState = (userId: number): void => {
  setState(prevState => {
    const newState = new Set(prevState);
    newState.add(userId);
    return newState;
  });
};
 ```

This ensures that you are not mutating the previous state directly and that React will properly update the component when the state changes.